modules/homa-lite/src/lib.rs

```diff
@@ -256,5 +276,304 @@ pub mod module {
 			Self::deposit_event(Event::<T>::XcmDestWeightSet(xcm_dest_weight));
 			Ok(())
 		}
+
+		/// Put in an request to redeem Staking currencies used to mint Liquid currency.
+		/// The redemption will happen after the currencies are unbounded on the relaychain.
+		///
+		/// Parameters:
+		/// -
+		#[pallet::weight(< T as Config >::WeightInfo::request_redeem())]
+		#[transactional]
+		pub fn request_redeem(origin: OriginFor<T>, liquid_amount: Balance, additional_fee: Permill) -> DispatchResult {
+			let who = ensure_signed(origin)?;
+
+			if liquid_amount.is_zero() {
+				// If the maount is zero, cancel previuos redeem request.
+				if let Some((_, _)) = Self::redeem_requests(&who) {
+					// Unreserve the liquid fee and remove the redeem request.
+					ensure!(
+						T::Currency::unreserve(T::LiquidCurrencyId::get(), &who, liquid_amount) == liquid_amount,
```

Comment:
```suggestion
						T::Currency::unreserve(T::LiquidCurrencyId::get(), &who, liquid_amount).is_zero(),
```
the return value is how much failed to unreserve